imagebuildah/build.go

```diff
@@ -340,6 +340,32 @@ func buildDockerfilesOnce(ctx context.Context, store storage.Store, logger *logr
 
 	warnOnUnsetBuildArgs(logger, mainNode, options.Args)
 
+	// --platform was explicitly selected for this build
+	// so set correct TARGETPLATFORM in args if it is not
+	// already selected by the user.
+	if options.SystemContext.OSChoice != "" && options.SystemContext.ArchitectureChoice != "" {
+		// os component from --platform string populates TARGETOS
+		if _, ok := options.Args["TARGETOS"]; !ok {
+			options.Args["TARGETOS"] = options.SystemContext.OSChoice
+		}
+		// arch component from --platform string populates TARGETARCH
+		if _, ok := options.Args["TARGETARCH"]; !ok {
+			options.Args["TARGETARCH"] = options.SystemContext.ArchitectureChoice
+		}
+		// variant component from --platform string populates TARGETVARIANT
+		if _, ok := options.Args["TARGETVARIANT"]; !ok {
+			if options.SystemContext.VariantChoice != "" {
+				options.Args["TARGETVARIANT"] = options.SystemContext.VariantChoice
+			}
+		}
+		if _, ok := options.Args["TARGETPLATFORM"]; !ok {
+			options.Args["TARGETPLATFORM"] = options.SystemContext.OSChoice + "/" + options.SystemContext.ArchitectureChoice
+			if options.SystemContext.VariantChoice != "" {
```

Comment:
if options.Args["TARGETVARIANT"] != "" {

Reply:
this is same as above since we are using `options.SystemContext.VariantChoice` to populate and not `buildargs.